fs/indexed-pkg-importer/src/index.ts

```diff
@@ -27,6 +27,12 @@ function createImportPackage (packageImportMethod?: 'auto' | 'hardlink' | 'copy'
     packageImportMethodLogger.debug({ method: 'hardlink' })
     return hardlinkPkg.bind(null, linkOrCopy)
   case 'auto': {
+    // Set default import method to hardlinking on Windows
+    if (process.platform === 'win32') {
+      packageImportMethodLogger.debug({ method: 'hardlink' })
+      return hardlinkPkg.bind(null, linkOrCopy)
+    }
+
```

Comment:
It might be not always possible to create a hard link. For instance, if the store is not on the same drive as the node_modules. It should still fall back to copying in that case.

Reply:
Maybe I should revert the refactor and use [74f770e](https://github.com/pnpm/pnpm/pull/7564/commits/74f770e57a29461657895996e2c2c3b570d79fa8)?